x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmGenerateTests.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.jose.jwk.OctetSequenceKey;
+import com.nimbusds.jwt.SignedJWT;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.MockSecureSettings;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.SettingsException;
+import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.xpack.core.security.authc.RealmConfig;
+import org.elasticsearch.xpack.core.security.authc.RealmSettings;
+import org.elasticsearch.xpack.core.security.authc.jwt.JwtRealmSettings;
+import org.elasticsearch.xpack.core.security.authc.support.UserRoleMapper;
+import org.elasticsearch.xpack.core.security.user.User;
+import org.elasticsearch.xpack.core.ssl.SSLService;
+
+import java.nio.charset.StandardCharsets;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class JwtRealmGenerateTests extends JwtRealmTestCase {
+    private static final Logger LOGGER = LogManager.getLogger(JwtRealmGenerateTests.class);
+
+    /**
+     * Generate a static JWT realm config, and request headers, which the realm can successfully authenticate.
+     *
+     * Print these artifacts, so they can be imported into this integration test project for verifying loading of JWT realm config:
+     *  - x-pack/plugin/security/qa/smoke-test-all-realms/build.gradle
+     *
+     * If something needs tweaking, change the code here and re-run it. The new artifacts can be used to update that smoke test project.
+     * @throws Exception Error if something goes wrong.
+     */
+    public void testCreateJwtSmokeTestRealm() throws Exception {
```

Comment:
My previous suggestion is that this method is not a test (because it does not test anything) and hence is better to be a helper class/method instead of running as part of the test suite. I now notice how it utilize the super class methods overall the places. It's not easy to relocate them and probably not worth it. I am happy for this to stay as is.